sktime/transformations/series/difference.py

```diff
@@ -0,0 +1,315 @@
+            "a list, tuple or np.ndarray of positive integers,"
+            f"but found {type(lags)}.",
+        ]
+    )
+    non_positive_msg = "`lags` should be positive integers."
+    if isinstance(lags, int):
+        if lags <= 0:
+            raise ValueError(non_positive_msg)
+        lags = check_array([lags], ensure_2d=False)
+    elif isinstance(lags, (list, tuple, np.ndarray)):
+        if not all([is_int(lag) for lag in lags]):
+            raise TypeError(msg)
+        lags = check_array(lags, ensure_2d=False)
+        if (lags <= 0).any():
+            raise ValueError(non_positive_msg)
+    else:
+        raise TypeError(msg)
+
+    return lags
+
+
+def _diff_transform(Z, lags):
+    Zt = Z.copy()
+
+    if len(lags) == 0:
+        return Zt
+
+    else:
+        for lag in lags:
+            Zt = Zt.diff(lag)
+        return Zt
+
+
+def _inverse_diff(Z, lag):
+    for i in range(lag):
+        Z.iloc[i::lag] = Z.iloc[i::lag].cumsum()
+
+    return Z
+
+
+class Differencer(_SeriesToSeriesTransformer):
+    """Apply iterative differences to a timeseries.
+
+    The transformation works for univariate and multivariate timeseries. However,
+    the multivariate case applies the same differencing to every series.
+
+    Difference transformations are applied at the specified lags in the order
+    provided.
+
+    For example, given a timeseries with monthly periodicity, using lags=[1, 12]
+    corresponds to applying a standard first difference to handle trend, and
+    followed by a seasonal difference (at lag 12) to attempt to account for
+    seasonal dependence.
+
+    To provide a higher-order difference at the same lag list the lag multiple
+    times. For example, lags=[1, 1] takes iterative first differences like may
+    be needed for a series that is integrated of order 2.
+
+    Parameters
+    ----------
+    lags : int or array-like, default = 1
+        The lags used to difference the data.
+        If a single `int` value is
+
+    remove_missing : bool, default = True
+        Whether the differencer should remove the initial observations that
+        contain missing values as a result of the differencing operation(s).
+
+    Attributes
+    ----------
+    lags : int or array-like
+        Lags used to perform the differencing of the input series.
+
+    remove_missing : bool
+        Stores whether the Differencer removes initial observations that contain
+        missing values as a result of the differencing operation(s).
+
+    Example
+    -------
+    >>> from sktime.transformations.series.difference import Differencer
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> transformer = Differencer(lags=[1, 12])
+    >>> y_transform = transformer.fit_transform(y)
+    """
+
+    _tags = {
+        "fit-in-transform": False,
+        "transform-returns-same-time-index": False,
+        "univariate-only": False,
+    }
+
+    def __init__(self, lags=1, remove_missing=True):
```

Comment:
in line with common usage in pandas
```suggestion
    def __init__(self, lags=1, drop_na=True):
```

Reply:
I'm down for that. Will push updated version shortly.